pkg/filestore/s3/s3.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/filestore"
+)
+
+type Store struct {
+	client *s3.S3
+	bucket string
+
+	logger *zap.Logger
+}
+
+type Option func(*Store)
+
+func WithLogger(logger *zap.Logger) Option {
+	return func(s *Store) {
+		s.logger = logger.Named("s3")
+	}
+}
+
```

Comment:
nit: The credential file is optional right? So I feel like it's better to set the path to the credentials file via `Option` like:

```go
func WithCredentialsFile(path string) Option {
	return func(s *Store) {
		s.credentialsFile = path
	}
}
```

Just want to unify with other parts.

Reply:
addressed on https://github.com/pipe-cd/pipe/pull/1485/commits/065211a7f92aefe59cdf7fba4255e1a3a67bca6e 🙏